### DISCUSSION TITLE
【需求】【预期的表现】能否增加一些的 AE自带”曲线/曝光度“基础插件的功能

### DISCUSSION THREAD

**Discussion by mianma12138** [Tencent/libpag#2364] [Ideas] [answered]
【预期的表现】能否增加一些的 AE自带”曲线/曝光度“基础插件的功能，非bmp合成中使用。
看了下官方文档支持的一些自带AE插件，其中缺少两个最常用的插件-曲线和曝光度，希望pag能够直接支持这两个效果的输出。
希望pag越来越完善。
![image](https://github.com/Tencent/libpag/assets/92774467/5f6ba12d-5792-4113-9442-f9b3b72a2a77)

**Comment by kevingpqi123** [comment]
能否简要说下这里的应用场景以及附上该效果的 AE 工程，我们会根据类似需求的重要性来调整支持的优先级

**Reply by mianma12138** [reply, marked as answer]
您好管理员，这个建议的原由是：曲线、曝光度这两个AE自带效果，使用频率极高，与目前pag支持效果列表中的“色相/饱和度，亮度对比度，高斯模糊”这类高频基础插件类似，希望可以支持。从目前测试的一些文件来看，使用曲线或者曝光度的AE合成必须通过_bmp来解决，其他自带效果也可代替，但不能完全替代。
不知道这种优化麻不麻烦。
[AE自带效果演示文件夹.zip](https://github.com/user-attachments/files/16659141/AE.zip)

**Reply by mianma12138** [reply]
另外，还有一个急需优化的点：长文本强制垂直居中。希望能取消强制居中或者由选项设置。
https://github.com/Tencent/libpag/discussions/2419#discussioncomment-10335266

**Reply by kevingpqi123** [reply]
关于文本，问题点在于 AE 中没有居中对齐，只有顶部对齐，但是对于大部分业务场景，需要的都是居中对齐，因此这里将顶部对齐设置成了居中对齐，除了顶部对齐，其它对齐方式在业务侧可以通过 API 设置
